### PR TITLE
Handle dao state conflicts better

### PR DESCRIPTION
**Pull request by ManfredKarrer**
- Check if conflicting node is seed node or not. Show info in list entry
and only show resync button and red color in text if conflict is with
seed node.
- At resync move all p2p network data to a dedicated directory so they
will be reloaded from the seeds.

### CODE REVIEW

**Comment by ManfredKarrer** [comment]
@sqrrm @ripcurlx 
I am not sure if we shoud merge that into 1.0.0 or keep it for later. It is not high risk but also not trivial to test. I tested quite a bit though... It should not introduce much risk as well.
Lets discuss if we should add it or not. Would at least reduce confusion if some peers are out of sync as we have seen in dao-regtest last days... Also the missing file deletion of proposals, blind votes and ballots would not have lead to a recovery in current release if the cause for the issue is that one of those db files is out of sync.

File: common/src/main/java/bisq/common/storage/FileManager.java [hunk 1]
```diff
@@ -162,16 +162,21 @@ void shutDown() {
...
             if (!corruptedBackupDir.mkdir())
                 log.warn("make dir failed");
 
-        File corruptedFile = new File(Paths.get(dir.getAbsolutePath(), "backup_of_corrupted_data", fileName).toString());
+        File corruptedFile = new File(Paths.get(dbDir.getAbsolutePath(), backupFolderName, fileName).toString());
         FileUtil.renameFile(storageFile, corruptedFile);
     }
 
+    public synchronized void removeAndBackupFile(String fileName) throws IOException {
+        removeAndBackupFile(storageFile, dir, fileName, "backup_of_corrupted_data");
```
Comment (sqrrm) [inline common/src/main/java/bisq/common/storage/FileManager.java]: The argument order is `dbDir`, `storageFile`..., this looks inverted.

File: desktop/src/main/java/bisq/desktop/main/dao/monitor/StateMonitorView.java [hunk 2]
```diff
@@ -102,11 +115,15 @@
     public StateMonitorView(DaoStateService daoStateService,
                             DaoFacade daoFacade,
                             CycleService cycleService,
-                            PeriodService periodService) {
+                            PeriodService periodService,
+                            SeedNodeRepository seedNodeRepository,
+                            File storageDir) {
```
Comment (sqrrm) [inline desktop/src/main/java/bisq/desktop/main/dao/monitor/StateMonitorView.java]: Should this be `@Named(Storage.STORAGE_DIR) File storageDir`?

**Review by sqrrm — COMMENTED** [review]
I think it's worth adding this before 1.0 as it would avoid a lot of extra support issues from conflicted non seed nodes. It's also likely that users will be able to recover through the UI with this resync from seed feature.

No ACK until my comments are taken care of though.

File: desktop/src/main/java/bisq/desktop/main/dao/monitor/StateMonitorView.java (hunk 2 above)
Comment (ManfredKarrer) [reply desktop/src/main/java/bisq/desktop/main/dao/monitor/StateMonitorView.java]: No its the base class, I will remove the Inject annotation and make it protected.

File: common/src/main/java/bisq/common/storage/FileManager.java (hunk 1 above)
Comment (ManfredKarrer) [reply common/src/main/java/bisq/common/storage/FileManager.java]: Good finding! Yes that is wrong. Thanks!

**Comment by sqrrm** [comment]
utACK

This version I think is worth merging before 1.0, it's not that complex and much time will be saved by not having users worry about client not in synch.

**Pull request merged**